File: app/javascript/factories/notes.ts
import faker from "faker";
import { Factory } from "fishery";
import { Note } from "models/Note";

export default Factory.define<Note>(({ sequence }) => ({
  id: sequence,
  title: faker.lorem.sentence(),
  created_at: faker.date.past().toISOString(),
  updated_at: faker.date.past().toISOString(),
}));

File: spec/factories/notes.rb
# frozen_string_literal: true

# == Schema Information
#
# Table name: notes
#
#  id         :bigint           not null, primary key
#  body       :string
#  title      :string
#  created_at :datetime         not null
#  updated_at :datetime         not null
#  user_id    :bigint           not null
#
# Indexes
#
#  index_notes_on_user_id  (user_id)
#
# Foreign Keys
#
#  fk_rails_...  (user_id => users.id)
#
FactoryBot.define do
  factory :note do
    title { Faker::Lorem.sentence }
    created_at { Faker::Date.backward(days: 14) }
    updated_at { Faker::Date.backward(days: 14) }
  end
end
